File: clan/2.c
#include <stdio.h>
#include <math.h>

typedef unsigned long long ull;

const double gl = (1 + sqrt(5))/2;

const double z = (1 - sqrt(5))/2;

double fibn(ull n) {
    return (pow(gl,n) - pow(z, n))/ sqrt(5);
}

int main() {
  double n = 4000000;

  ull ind = 3;
  double sum = 0;
  double cur = fibn(ind);
  while (cur < n) {
    sum += cur;
    ind += 3;
    cur = fibn(ind);
  }

  printf("%f\n", sum);
}

File: rust/2.rs
use std::io::{self};

fn fibn(n:f64) -> f64 {
  (gl.powf(n) - z.powf(n))/5.0_f64.sqrt()
}

fn main() {
  let n:f64 = 40.0;
  const gl:f64 = (1.0 + 5.0_f64.sqrt())/2.0;

  const z:f64 = (1.0 - 5.0_f64.sqrt())/2.0;

  let mut ind:f64 = 3.0;
  let mut sum:f64 = 0.0;
  let mut cur:f64 = fibn(ind);

  while (cur < n) {
    sum += cur;
    ind += 3.0;
    cur = fibn(ind);
  }

  println!("{}", sum);
}
